cranelift/codegen/src/isa/x64/abi.rs

```diff
@@ -280,7 +280,7 @@ impl ABIBody for X64ABIBody {
                     return Inst::mov_r_r(/*is64=*/ true, from_reg.to_reg(), to_reg);
                 } else if from_reg.get_class() == RegClass::V128 {
                     // TODO: How to support Movss. Should is64 always be true?
-                    return Inst::xmm_r_r(SSE_Op::SSE2_Movsd, from_reg.to_reg(), to_reg);
+                    return Inst::xmm_r_r(SseOpcode::Movsd, from_reg.to_reg(), to_reg);
```

Comment:
Ahh ... this renaming loses the SSE-level on the name; I specifically wanted to retain that -- per discussion with Johnny in PR1665.  Can you reinstate it somehow?  I think it will be confusing and difficult -- especially later, when we add even more opcodes (and there are zillions of them) -- if we cannot look at an SseOpcode value and know directly what level it is for.

Reply:
I see from reading later that we can extract the SSE level by passing calling its `.available_from` method.  But I don't see the win here .. it places more of a burden on readers of the code, who now have to look in some other place to figure out the SSE level.

Reply:
Sorry, i should have commented a bit in the commit message. With the `available_from` function, we can then automate the checks that the instructions are available when comparing to a given SSE level. So this could be used to dynamically (at runtime) check that we only use allowed instructions, as well as selecting instructions based on the real SSE level.

So it's slightly more indirection, but with this check (to come in a subsequent pull request), there's no way it can be misused. In terms of developer experience, in most modern editors, this is a "jump to definition" then grep for "available_from", which is not too bad (and should not be done so often). Overall I found better to have something that could be probed and checked programmatically rather than a "visual" check that's subject to operators errors :)

With this new information, do you agree with the current form, or do you strongly feel about reverting it?

Reply:
Discussed on irc and Julian agreed with this, resolving.